src/__tests__/index.browser.js

```diff
@@ -5,45 +5,94 @@
  */
 
 import test from 'tape-cup';
-import RPC from '../browser';
 
-test('success status request', t => {
-  const fetch = (...args) =>
+import App, {createPlugin} from 'fusion-core';
+import {FetchToken, createToken} from 'fusion-tokens';
+import {getSimulator} from 'fusion-test-utils';
+
+import {RPCHandlersToken} from '../tokens';
+import RPCPlugin from '../browser';
+
+const MockPluginToken = createToken('test-plugin-token');
+function createTestFixture() {
+  const mockFetch = (...args) =>
     Promise.resolve({json: () => ({status: 'success', data: args})});
-  const rpc = RPC({fetch})();
-  t.equals(typeof rpc.request, 'function', 'has method');
-  t.ok(rpc.request('test') instanceof Promise, 'has right return type');
-  rpc
-    .request('test')
-    .then(([url, options]) => {
-      t.equals(url, '/api/test', 'has right url');
-      t.equals(options.method, 'POST', 'has right http method');
-      t.equals(
-        options.headers['Content-Type'],
-        'application/json',
-        'has right content-type'
-      );
-      t.equals(options.body, '{}', 'has right body');
-      t.end();
+  const mockHandlers = {};
+
+  const app = new App('content', el => el);
+  app.register(FetchToken, mockFetch);
+  app.register(RPCHandlersToken, mockHandlers);
+  app.register(MockPluginToken, RPCPlugin);
+  return app;
+}
+
+test('success status request', t => {
+  const app = createTestFixture();
+
+  let wasResolved = false;
+  getSimulator(
+    app,
+    createPlugin({
```

Comment:
I didn't realize we supported passing plugins to `getSimulator`. What is the purpose of this vs just calling `app.register` ?

Reply:
No tangible difference.  It's just to explicitly denote that the "plugin" there is for testing only.